cms/static/js/views/pages/container.js

```diff
@@ -939,14 +922,31 @@ function($, _, Backbone, gettext, BasePage,
 
             placeholderElement = $placeholderEl.appendTo($container);
 
-            if (this.options.isIframeEmbed) {
-                if (iframeMessageData.payload.data && iframeMessageData.type === 'addXBlock') {
-                    return this.onNewXBlock(placeholderElement, scrollOffset, false, iframeMessageData.payload.data);
-                }
+            if (this.options.isIframeEmbed && !this.isSplitTestContentPage) {
+              if (iframeMessageData.payload.data && iframeMessageData.type === 'addXBlock') {
+                  return this.onNewXBlock(placeholderElement, scrollOffset, false, iframeMessageData.payload.data);
+              }
+            }
+
+            if (this.options.isIframeEmbed && this.isSplitTestContentPage) {
+                this.postMessageToParent({
+                    type: 'addNewComponent',
+                    message: 'Add new XBlock',
+                    payload: {},
+                });
             }
 
             return $.postJSON(this.getURLRoot() + '/', requestData,
                 _.bind(this.onNewXBlock, this, placeholderElement, scrollOffset, false))
+                .success(function () {
+                    if (self.options.isIframeEmbed && self.isSplitTestContentPage) {
+                        self.postMessageToParent({
+                            type: 'hideProcessingNotification',
+                            message: 'Hide processing notification',
+                            payload: {}
+                        });
+                    }
+                })
                 .fail(function() {
```

Comment:
[question]: Should ProcessingNotification be hidden in case of an error?

Reply:
Sure, added thanks.

Reply:
Resolved